src/main/java/com/mate/fruitshop/strategy/ProcessPurchaseTransaction.java

```diff
@@ -0,0 +1,24 @@
+package com.mate.fruitshop.strategy;
+
+import com.mate.fruitshop.dao.FruitStorageDao;
+import com.mate.fruitshop.dao.impl.FruitStorageDaoImpl;
+import com.mate.fruitshop.model.FruitEntry;
+import com.mate.fruitshop.model.Transaction;
+
+public class ProcessPurchaseTransaction implements TransactionProcessingStrategy {
+    private final FruitStorageDao dao;
+
+    public ProcessPurchaseTransaction() {
+        this.dao = new FruitStorageDaoImpl();
+    }
+
+    @Override
+    public boolean process(Transaction transaction) {
+        FruitEntry fruitEntry = dao.getEntryByFruitName(transaction.getFruitName());
+        if (fruitEntry != null) {
+            fruitEntry.setQuantity(fruitEntry.getQuantity() - transaction.getQuantity());
+            return true;
+        }
+        return false;

```

Comment:
Well, such ignoring of obvious errors in input data is unacceptable)
We definitely have to inform our user about "successfull" fruits purchasing operation from empty storage.
Think of it here and in other transactions.
Or have i missed processing of this method returning false?